teuthology/orchestra/remote.py

```diff
@@ -218,6 +218,47 @@ def mktemp(self):
         data = proc.stdout.getvalue()
         return data
 
+    def sh(self, script, **kwargs):
+        """
+        Shortcut for run method.
+
+        Usage:
+            my_name = remote.sh('whoami')
+            remote_date = remote.sh('date')
+        """
+        if 'stdout' not in kwargs:
+            kwargs['stdout'] = StringIO()
+        if 'args' not in kwargs:
+            kwargs['args'] = script
+        proc=self.run(**kwargs)
+        return proc.stdout.getvalue()
+
+
+    def sh_file(self, script, label="script", sudo=False, **kwargs):
+        """
+        Run shell script after copying its contents to a remote file
+
+        :param script:  string with script text, or file object
+        :param sudo:    run command with sudo if True,
+                        run as user name if string value (defaults to False)
+        :param label:   string value which will be part of file name
+        Returns: stdout
+        """
+        ftempl = '/tmp/teuthology-remote-$(date +%Y%m%d%H%M%S)-{}-XXXX'\
```

Comment:
Why the XXXX at the end there?
(Rest of this looks good to me!)

Reply:
quote from MKTEMP(1)

> SYNOPSIS
>        mktemp [OPTION]... [TEMPLATE]
> 
> DESCRIPTION
>        Create  a  temporary  file or directory, safely, and print its name.  TEMPLATE must contain at least 3 consecutive 'X's in last component.  If TEMPLATE is not specified, use tmp.XXXXXXXXXX, and --tmpdir is implied.